rust/agents/processor/src/processor.rs

```diff
@@ -391,6 +391,10 @@ impl NomadAgent for Processor {
         Self: Sized + 'static,
     {
         tokio::spawn(async move {
+            if self.is_home_failed().await?? {
+                return Err(Report::new(BaseError::FailedHome));
```

Comment:
Can replace all usage of `return Err(Report::new(ErrorEnum::SpecificError))` with `return Err(ErrorEnum::SpecificError.into())`. See example [here](https://github.com/nomad-xyz/nomad-monorepo/blob/main/rust/agents/updater/src/produce.rs#L56-L60)

Reply:
Agree! going to come in the next commit